packages/apollo-language-server/src/config/__tests__/loadConfig.ts

```diff
@@ -0,0 +1,239 @@
+};
+
+const writeFilesToDir = (dir: string, files: Record<string, string>) => {
+  Object.keys(files).forEach(key => {
+    if (key.includes("/")) makeNestedDir(path.dirname(key));
+    fs.writeFileSync(`${dir}/${key}`, files[key]);
+  });
+};
+
+describe("loadConfig", () => {
+  let dir, dirPath;
+
+  // set up a temp dir
+  beforeEach(() => {
+    dir = fs.mkdtempSync("__tmp__");
+    dirPath = `${process.cwd()}/${dir}`;
+  });
+
+  // clean up our temp dir
+  afterEach(() => {
+    if (dir) deleteFolderRecursive(dir);
+    dir = dirPath = undefined;
+  });
+
+  it("loads with client defaults from different dir", async () => {
+    writeFilesToDir(dir, {
+      "my.config.js": `
+        module.exports = {
+          client: {
+            service: 'hello'
+          }
+        }
+      `
+    });
+
+    const config = await loadConfig({
+      configPath: dirPath,
+      configFileName: "my.config.js"
+    });
+    expect(config.rawConfig).toMatchInlineSnapshot(`
+Object {
+  "client": Object {
+    "addTypename": true,
+    "clientOnlyDirectives": Array [
+      "connection",
+      "type",
+    ],
+    "clientSchemaDirectives": Array [
+      "client",
+      "rest",
+    ],
+    "excludes": Array [
+      "**/node_modules",
+      "**/__tests__",
+    ],
+    "includes": Array [
+      "src/**/*.{ts,tsx,js,jsx,graphql}",
+    ],
+    "service": "hello",
+    "statsWindow": Object {
+      "from": -86400,
+      "to": -0,
+    },
+    "tagName": "gql",
+  },
+  "engine": Object {
+    "endpoint": "https://engine-graphql.apollographql.com/api/graphql",
+    "frontend": "https://engine.apollographql.com",
+  },
+}
+`);
+  });
+
+  it("loads with service defaults from different dir", async () => {
+    writeFilesToDir(dir, {
+      "my.config.js": `
+        module.exports = {
+          service: {
+            name: 'hello'
+          }
+        }
+      `
+    });
+
+    const config = await loadConfig({
+      configPath: dirPath,
+      configFileName: "my.config.js"
+    });
+    expect(config.rawConfig).toMatchInlineSnapshot(`
+Object {
+  "engine": Object {
+    "endpoint": "https://engine-graphql.apollographql.com/api/graphql",
+    "frontend": "https://engine.apollographql.com",
+  },
+  "service": Object {
+    "endpoint": Object {
+      "url": "http://localhost:4000/graphql",
+    },
+    "excludes": Array [
+      "**/node_modules",
+      "**/__tests__",
+    ],
+    "includes": Array [
+      "src/**/*.{ts,tsx,js,jsx,graphql}",
+    ],
+    "name": "hello",
+  },
+}
+`);
+  });
+
+  it("[deprecated] loads config from package.json", async () => {
+    writeFilesToDir(dir, {
+      "package.json": `{"apollo":{"client": {"service": "hello"}} }`
+    });
+    const config = await loadConfig({ configPath: dirPath });
+
+    expect(config.client.service).toEqual("hello");
+  });
+
+  it("loads config from a ts file", async () => {
+    writeFilesToDir(dir, {
+      "apollo.config.ts": `module.exports = {"client": {"service": "hello"}`
```

Comment:
Hmmm... I haven't seen vscode mess up _yet_ because of this. I wanted to leave at least one test with the default filenames just to make sure those are working properly as well

Reply:
It'll still probably trigger a project reload but 1 is much more palatable than 50. I'm good with leaving one in to be sure!

Reply:
would the `.js` and `package.json` files also trigger a reload?

Reply:
Specifically `apollo.config.js` does, but I think I left out support for `package.json` intentionally when I did the work to reload projects on config update.